tests/script_runner/parameter_with_complex_types.py

```diff
@@ -0,0 +1,34 @@
+import sys
+from typing import Optional, Union
+
+from hera.shared import global_config
+from hera.workflows import script
+
+global_config.experimental_features["script_annotations"] = True
+
+
+@script(constructor="runner")
+def optional_str_parameter(my_string: Optional[str]) -> Optional[str]:
```

Comment:
We should make the change referred in https://github.com/argoproj-labs/hera/pull/1147#issuecomment-2298303536 to make this an error. We shouldn't add tests without default values as it needs changing anyway

Reply:
Added default value!

Reply:
Could we add a test and error handling for when a default value is not present?

Reply:
Added!